### PR TITLE
[api-extractor] support .d.mts and .d.cts files

### PR DESCRIPTION
**Pull request by ds300**
## Summary

TypeScript supports emitting and loading typings with `.d.cts` and `.d.mts` file extensions. We published a package which uses these extensions for its typings file, but it broke our api-extractor usage so we applied this patch.

## Details

This was the only place I found that specifically hard-coded the `.d.ts` requirement.

## How it was tested

Sorry I don't have time to dig into how you're doing testing in this repo.

## Impacted documentation

I don't think this affects any docs, or needs to be explicitly document itself.

### CODE REVIEW

File: apps/api-extractor/src/api/ExtractorConfig.ts [hunk 1]
```diff
@@ -210,7 +210,7 @@ export class ExtractorConfig {
     path.join(__dirname, '../schemas/api-extractor-defaults.json')
   );
 
-  private static readonly _declarationFileExtensionRegExp: RegExp = /\.d\.ts$/i;
```
Comment (dmichon-msft) [inline apps/api-extractor/src/api/ExtractorConfig.ts:213]: TypeScript 5 is also adding the ability to reference `.d.*.ts` for typings for non-js files, with `.d.js.ts` technically a valid alternative to standard `.d.ts`. We may want to add that consideration here as well.

Comment (ds300) [reply apps/api-extractor/src/api/ExtractorConfig.ts:213]: Do you want me to do that?

**Comment by JoshuaKGoldberg** [comment]
👋 friendly ping @dmichon-msft, is there anything we can do to get this merged in?

File: apps/api-extractor/src/api/ExtractorConfig.ts (hunk 1 above)
Comment (dmichon-msft) [reply apps/api-extractor/src/api/ExtractorConfig.ts:213]: We can deal with that separately.

**Review by dmichon-msft — APPROVED** [review]

**Comment by JoshuaKGoldberg** [comment]
cc ... @iclanton? Is there someone we can bug about this? 🙂

**Comment by JoshuaKGoldberg** [comment]
🙌 thanks!

**Pull request merged**